client/cluster.go

```diff
@@ -411,10 +413,25 @@ func (c *ClusterClient) DrainNode(name string) error {
 func (c *ClusterClient) NodeExists(name string) (bool, error) {
 	c.logger.Info("client_node_exists", zap.String("name", name))
 
-	if err := c.kubectl.NodeExists(name); err != nil {
+	addPrefix := true
+
+	if c.cluster.CloudProvider() == api.AWS {
+		myvar, err := c.Machine(name)
```

Comment:
This variable should have a more descriptive name

```suggestion
		machine, err := c.Machine(name)
```